File: src/main/java/org/cascadebot/cascadebot/utils/votes/VoteFinishFunction.java
package org.cascadebot.cascadebot.utils.votes;

import net.dv8tion.jda.api.entities.TextChannel;

import java.util.List;
import java.util.function.BiConsumer;

public enum VoteFinishFunction {
    DUMMY(((textChannel, voteResults) -> {

    }));

    private final BiConsumer<TextChannel, List<VoteResult>> consumer;
    VoteFinishFunction(BiConsumer<TextChannel, List<VoteResult>> consumer) {
        this.consumer = consumer;
    }

    public BiConsumer<TextChannel, List<VoteResult>> getConsumer() {
        return consumer;
    }
}

File: src/main/kotlin/org/cascadebot/cascadebot/utils/votes/VoteFinishFunction.kt
/*
 * Copyright (c) 2022 CascadeBot. All rights reserved.
 * Licensed under the MIT license.
 */

package org.cascadebot.cascadebot.utils.votes

import net.dv8tion.jda.api.entities.TextChannel

enum class VoteFinishFunction(val function: (TextChannel, List<VoteResult>) -> Unit) {
    DUMMY({ _, _ -> });
}